src/dune_rules/inline_tests.ml

```diff
@@ -170,19 +170,19 @@ include Sub_system.Register_end_point (struct
           ~melange_package_name:None
           ~package
       in
-      let linkages =
-        let modes =
-          if Mode_conf.Set.mem info.modes Javascript
-          then Mode_conf.Set.add info.modes Byte
-          else info.modes
```

Comment:
I guess this logic has been wrong for a while

Reply:
Can we add a test demonstrating the effects of this removal ?

Reply:
```diff
File "test/blackbox-tests/test-cases/jsoo/inline-tests.t/run.t", line 1, characters 0-0:
------ test/blackbox-tests/test-cases/jsoo/inline-tests.t/run.t
++++++ test/blackbox-tests/test-cases/jsoo/inline-tests.t/run.t.corrected
File "test/blackbox-tests/test-cases/jsoo/inline-tests.t/run.t", line 8, characters 0-1:
 |Run inline tests using node js
 |
 |  $ cat >dune-project <<EOF
 |  > (lang dune 2.6)
 |  > EOF
 |
+|  $ dune build js/.inline_tests_js.inline-tests/inline_test_runner_inline_tests_js.bc --display short
+|  Error: Don't know how to build
+|  js/.inline_tests_js.inline-tests/inline_test_runner_inline_tests_js.bc
+|  [1]
 |
```

Reply:
Thanks for the test. I have added it.

Reply:
For the record, this incorrect logic was introduced in #5049+#6645